### ISSUE TITLE
Build errors with Intel compiler on ALCF Theta machine

### ISSUE THREAD

**Issue by tpeterka** [LLNL/Caliper#229]
Building with the default Intel compiler on the ALCF Theta machine results in the following error:

-----

[  8%] Building CXX object src/mpi-rt/controllers/CMakeFiles/caliper-mpi-controllers.dir/SpotController.cpp.o
In file included from /usr/include/c++/4.8/type_traits(35),
                 from /home/tpeterka/software/Caliper/include/caliper/common/Variant.h(15),
                 from /home/tpeterka/software/Caliper/include/caliper/common/Node.h(15),
                 from /home/tpeterka/software/Caliper/include/caliper/common/Attribute.h(14),
                 from /home/tpeterka/software/Caliper/include/caliper/Caliper.h(11),
                 from /home/tpeterka/software/Caliper/src/mpi-rt/controllers/SpotController.cpp(6):
/usr/include/c++/4.8/bits/c++0x_warning.h(32): error: #error directive: This file requires compiler and library support for the ISO C++ 2011 standard. This support is currently experimental, and must be enabled with the -std=c++11 or -std=gnu++11 compiler options.
  #error This file requires compiler and library support for the \
   ^

-----

More errors follow afterwards, presumably as a result.

My cmake configuration is as follows:
cmake .. \
-DCMAKE_INSTALL_PREFIX=/home/tpeterka/software/Caliper/install-gnu \
-DCMAKE_C_COMPILER=cc \
-DCMAKE_CXX_COMPILER=CC \
-DWITH_MPI=on \

I tried adding -DCMAKE_CXX_STANDARD=11 to my configuration, but it did not change anything.

**Comment by daboehme** [comment]
Hi @tpeterka ,

What cmake version are you using? Older cmake versions (before around 3.10 or so) didn't set the C++11 flag for Intel compilers correctly. You can try upgrading cmake, or explicitly setting the flag with `-DCMAKE_CXX_FLAGS="-std=c++11"`.

**Comment by jrmadsen** [comment]
Also, you may want to try `-DCMAKE_CXX_STANDARD_REQUIRED=ON` which will force CMake to append the flag.

**Comment by tpeterka** [comment]
I can confirm that upgrading cmake solved the problem. The default was version 3.5.2, and when I switched to 3.14.5, the build was successful.

Trying to get the 3.5.2 version to work with the additional -std=c++11 flag and forcing the flag to be required did not work. I got a different error than before:

-----

[ 11%] Building C object src/common/CMakeFiles/caliper-common.dir/cali_types.c.o
/home/tpeterka/software/Caliper/src/common/cali_types.c(40): error: expected an 
expression
    for (const struct typemap_t* t = typemap; t->str; ++t)
         ^

/home/tpeterka/software/Caliper/src/common/cali_types.c(40): error: identifier "
t" is undefined
    for (const struct typemap_t* t = typemap; t->str; ++t)
                                              ^

-----

I can easily switch modules to use the newer cmake. I only report the above behavior with the old cmake version for your reference.

Many thanks---Tom.
